checks/src/main/kotlin/com/jzbrooks/assertk/lint/checks/TestFrameworkAssertionDetector.kt

```diff
@@ -22,34 +24,180 @@ class TestFrameworkAssertionDetector :
+                        val messageExpr = node.valueArguments.firstOrNull()
+
+                        fix()
+                            .replace()
+                            .reformat(true)
+                            .range(
+                                context.getCallLocation(
+                                    node,
+                                    includeReceiver = false,
+                                    includeArguments = true,
+                                ),
+                            ).with(
+                                buildString {
+                                    append("assertk.fail(")
+                                    if (messageExpr != null) {
+                                        append(messageExpr.sourcePsi!!.text)
+                                    }
+                                    append(")")
+                                },
+                            ).build()
+                    }
+
+                    else -> null
+                }
+
+            private fun replaceJunit4AssertionWithExpected(
+                call: UCallExpression,
+                assertionFunctionName: String,
+                expectedTransformation: StringBuilder.(
+                    UExpression,
+                ) -> Unit = { append(it.sourcePsi!!.text) },
+            ): LintFix? {
+                val expectedIndex = if (call.valueArguments.size == 2) 0 else 1
+                val expectedExpr =
+                    call.valueArguments.getOrNull(expectedIndex) ?: return null
+                val actualExpr =
+                    call.valueArguments.getOrNull(expectedIndex + 1) ?: return null
+
+                return fix()
+                    .replace()
+                    .reformat(true)
+                    .range(
+                        context.getCallLocation(
+                            call,
+                            includeReceiver = false,
+                            includeArguments = true,
+                        ),
+                    ).imports("assertk.assertThat", "assertk.assertions.$assertionFunctionName")
+                    .with(
+                        buildString {
+                            append("assertThat(")
+                            append(actualExpr.sourcePsi!!.text)
+                            append(").")
+                            append(assertionFunctionName)
+                            append("(")
+                            expectedTransformation(expectedExpr)
+                            append(")")
+
+                            if (call.valueArguments.size == 3) {
+                                append(" // ")
+                                append(
+                                    call.valueArguments
+                                        .first()
+                                        .sourcePsi!!
+                                        .text,
+                                )
+                            }
```

Comment:
It could be worth offering this in a variation in case people don't really care to keep the message. Sometimes those messages carry useful context.

If assertk implements https://github.com/willowtreeapps/assertk/issues/534, then a variation based on that API would probably be useful.

For now, comments don't hurt and this approach errs on the side of not destroying information.